### ISSUE TITLE
Schedule: ajax behavior event for prev, next, today buttons

### ISSUE THREAD

**Issue by tuerker** [primefaces/primefaces#6313]
With each click on the previous, next, or today i want to re-render the p:schedule component because in my use case for instance the minTime/maxTime may change over the different days.

I have seen that currently the following five ajax behavior event are supported:

dateSelect | org.primefaces.event.SelectEvent | When a date is selected.
eventSelect | org.primefaces.event.SelectEvent | When an event is selected.
eventMove | org.primefaces.event.ScheduleEntryMoveEvent | When an event is moved.
eventResize | org.primefaces.event.ScheduleEntryResizeEvent | When an event is resized.
viewChange | org.primefaces.event.SelectEvent | When a view is changed.

It would be extremely needed to add at least one more event, which is triggered whenever prev, next, or today is clicked.

Or is there already a possibility implemented to realize this behavior?

Many thanks in advance.

**Comment by melloware** [comment]
I think these 3 buttons prev, next, and today should fire the `viewChange` event since that is similar to how Calendar and DatePicker do it.  I will fix.

**Comment by tuerker** [comment]
@melloware: thanks. this will do the job.

**Comment by tuerker** [comment]
@melloware. while fixing this issue, could be please check generally the triggering of the viewChange event. It looks like that the viewChange event is fired when i click on the day and month buttons but not when clicking on the week button. Thanks in advance.

**Comment by tuerker** [comment]
@melloware: forget my last comment. the viewChange event is of course only fired when you switch between the different views (day/week/month) and not on each click.

**Comment by melloware** [comment]
OK so I fixed this for 9.0.  If you need this right now in your project just add this JS to your page...
```javascript
$('.fc-prev-button, .fc-next-button, .fc-today-button').on('click.schedule', function() {
       PF('yourScheduleWidget').callBehavior('viewChange');
});
```

**Comment by tuerker** [comment]
Many thanks again. 

A quick question regarding the schedule component. I pulled the code from master of PF 9 and have seen that the schedule component must have changed seriously in the last weeks. With this code my schedule (see below) does not show the view buttons (day, week, month) anymore. Do i have to adapt something here?

    <h:panelGrid columnClasses="value">
        <p:schedule id="schedule"
                    value="#{instrumentReservationManager.lazyEventModel}"
                    initialDate="#{instrumentReservationManager.initialDate}"
                    showWeekends="#{instrumentReservationManager.instrumentReservationSetting.weekends}"
                    minTime="#{instrumentReservationManager.instrumentReservationSetting.minTimeFormat}"
                    maxTime="#{instrumentReservationManager.instrumentReservationSetting.maxTimeFormat}"
                    slotLabelInterval="#{instrumentReservationManager.instrumentReservationSetting.scheduleSlotLabelInterval}"
                    slotDuration="#{instrumentReservationManager.instrumentReservationSetting.scheduleSlotDuration}"
                    slotEventOverlap="false"
                    slotLabelFormat="HH:mm"
                    timeFormat="HH:mm"
                    columnHeaderFormat="week:'dddd DD MMM', month:'dddd'"
                    rightHeaderTemplate="timeGridDay,timeGridWeek,dayGridMonth"
                    leftHeaderTemplate="prev, next, today"
                    tooltip="true"
                    showWeekNumbers="false"
                    allDaySlot="false"
                    nextDayThreshold="00:00"
                    draggable="true"
                    resizable="true"
                    aspectRatio="100"
                    view="agendaDay"
                    extender="initSchedule">
          <p:ajax event="dateSelect"
                  listener="#{instrumentReservationManager.onDateSelect}"
                  disabled="#{!instrumentReservation.creatable}"
                  update="eventDetails"
                  oncomplete="PF('eventDialog').show();" />
          <p:ajax event="eventSelect"
                  listener="#{instrumentReservationManager.onEventSelect}"
                  disabled="#{!instrumentReservation.updatable}" />
          <p:ajax event="eventMove"
                  listener="#{instrumentReservationManager.onEventMove}"
                  update="messages" />
          <p:ajax event="eventResize"
                  listener="#{instrumentReservationManager.onEventResize}"
                  update="messages" />
          <p:ajax event="viewChange"
                  update="schedule" />
        </p:schedule>

**Comment by melloware** [comment]
PF 9 upgraded to FullCalendar 5.3.0 i would check out the Showcase from GitHub whcih is updated for 9.0 and all examples are working. Also check the 8.0-9.0 migration guide: https://github.com/primefaces/primefaces/wiki/Migration-Guide#80-to-90

**Comment by tuerker** [comment]
i will do. many thanks again.

**Comment by MatthiasScherrerFGCZ** [comment]
> OK so I fixed this for 9.0. If you need this right now in your project just add this JS to your page...
> 
> ```js
> $('.fc-prev-button, .fc-next-button, .fc-today-button').on('click.schedule', function() {
>        PF('yourScheduleWidget').callBehavior('viewChange');
> });
> ```

Thanks for the quick fix!
